### ISSUE TITLE
Do apps and other things build on airgram have to be GPL?

### ISSUE THREAD

**Issue by cluehaven** [airgram/airgram#121]
**Issue type:**

[x] question

**Airgram version:**
[x] `latest`

I hope this is a quick easy "dumb" question...

Airgram is used to make telegram (bot and/or client) apps and is kindly offered by the authors under GPL. Does it make it clear anywhere that building on top of airgram (e.g. a telegram client that uses it) is not considered derivative. I.e. it's only extending the actual airgram codebase that results in code which would need to be made public under GPL?

**Comment by 0x77dev** [comment]
It doesn't need to be licensed over GPL, you will need to mention an opensource libraries licenses if you will create a fork you will need to license it under GPL.

**Comment by Bannerets** [comment]
Yes, apps built on airgram have to be GPL. It is considered a combined work. LGPL was specially created to avoid this. GPL is very rarely used for libraries, even for those made by FSF. E.g. see https://www.gnu.org/licenses/gpl-faq.en.html#IfLibraryIsGPL, https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License#Differences_from_the_GPL.

Though, if you don't distribute your app, the source code doesn't have to be open. For example, if you created a Telegram bot that uses Airgram, and run it on a server without publising it, you aren't required to open source it. If you created a Telegram client and distribute it, users of the app must have access to the source code.

**Comment by cluehaven** [comment]
> Yes, apps built on airgram have to be GPL. It is considered a combined work. LGPL was specially created to avoid this. GPL is very rarely used for libraries. E.g. see https://www.gnu.org/licenses/gpl-faq.en.html#IfLibraryIsGPL, https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License#Differences_from_the_GPL.
> 
> Though, if you don't distribute your app, the source code doesn't have to be open. For example, if you created a Telegram bot that uses Airgram, and run it on a server without publising it, you aren't required to open source it. If you created a Telegram client and distribute it, users of the app must have access to the source code.

Very handy, thanks very much for the clear with references answer
